a6/app/serializers.py

```diff
@@ -0,0 +1,187 @@
+from rest_framework import serializers
+
+from .models import *
+
+class CategorySerializer(serializers.ModelSerializer):
+    #name = serializers.CharField(max_length=255)
+    class Meta:
+        model = Category
+        fields = ('id', 'name')
+
+class ProductSerializer(serializers.ModelSerializer):
+    code = serializers.CharField(max_length=20, required=True)
+    description = serializers.CharField(max_length=1000, required=True)
+    price = serializers.DecimalField(max_digits=15, decimal_places=5, required=True)
+    #id = serializers.IntegerField()
+    category_id = serializers.IntegerField(source='get_category_id', read_only=True)
+    category = serializers.CharField(max_length=255,write_only=True, required=True)
+    class Meta:
+        model = Product
+        fields = ('id', 'code', 'description', 'price', 'category_id', 'category', )
+
+    def create(self, validated_data):
+        code = validated_data['code']
+        description = validated_data['description']
+        price = validated_data['price']
+        category = validated_data['category']
+
+        # create category
+        try:
+            category_obj = Category.objects.filter(name=category).first()
+        except:
+            category_obj = Category(name=category)
+            category_obj.save()
+        # create product
+        product_obj = Product(code=code, description=description, price=round(float(price),4), deleted=0)
+        product_obj.save()
+        # add entry to categoryproduct
+        category_product_obj = CategoryProduct.objects.get_or_create(category=category_obj, product=product_obj)
+        #print self.__dict__
+        return product_obj
+
+    def update(self, instance, validated_data):
+        if not self.partial:
+            # PUT
+            code = validated_data['code'] if 'code' in validated_data else ''
```

Comment:
a cleaner way to do this is something like
instance.code = validated_data.get('code'), instance.code
please look into one of the serializers in 
https://github.com/priyanshuid/A06/pull/2/